### PR TITLE
Improvise Dynamic Routing

### PR DESCRIPTION
**Pull request by vjspranav**
* Remove Hash in the URL
* Also Improvise conditional rendering of splash

Fixes: #319

### CODE REVIEW

File: src/containers/Main.js
Comment (ashutosh1919) [inline src/containers/Main.js:57]: why do we need separate route especially for splash? Isn't having splash effect on enter good enough?

**Review by ashutosh1919 — CHANGES_REQUESTED** [review]
Also, please remove `package-lock.json` from the commit.

**Comment by ashutosh1919** [comment]
Screenshot 2023-10-03 at 5 35 38 PM

@vjspranav I don't think the routing logic is working correctly. please Look at the above ss. This happens when I do hot-reload.

File: src/containers/Main.js
Comment (vjspranav) [reply src/containers/Main.js:57]: I did not separate the route. If you see the commit, there was a if else that defined two sets of the same router with the only difference being. IF splash enabled a splash route was added and other not.
Seemed redundant, so just made it singular.

**Comment by vjspranav** [comment]
> Also, please remove `package-lock.json` from the commit.

Done

**Comment by vjspranav** [comment]
> Screenshot 2023-10-03 at 5 35 38 PM
> 
> @vjspranav I don't think the routing logic is working correctly. please Look at the above ss. This happens when I do hot-reload.

https://github.com/ashutosh1919/masterPortfolio/pull/320/files#diff-f934e43e0f8ddbcd51317189e847c41f190ec983462fc3bd7169b8d7f1a99cf6R25
If you can check the highlighted line, that would possibly be the issue, a variable needs to be changed from 0 to 1 depending on whether you are using a custom domain or a project page (with '/something')
for ref: vjspranav.dev (deployed using browser router)

**Comment by ashutosh1919** [comment]
@vjspranav the issue is still there. I didn't understand your point. Can you please describe in detail? Also, you can look at the deploy preview below this PR to see how your changes look.

**Comment by vjspranav** [comment]
@ashutosh1919 ?? It'd be nice to have this I feel

**Comment by ashutosh1919** [comment]
@vjspranav Sorry! We can't approve PRs unless the deploy preview shows the website properly.

Also, I want to understand if there is any greater advantage of using BrowserRouter?

Please feel free to close this PR if you think it is not fixable.

**Comment by vjspranav** [comment]
> @vjspranav Sorry! We can't approve PRs unless the deploy preview shows the website properly.
> 
> Also, I want to understand if there is any greater advantage of using BrowserRouter?
> 
> Please feel free to close this PR if you think it is not fixable.

@ashutosh1919 Hey fixing deploy preview isn't an issue, I was intentionally holding off on that so that you could actually once look at the spec around the PR I raise, but sure. I have pushed a commit, which fixes preview. (You can check now)

Two things here:
1. react-router-dom themselves do not recommend using Hash Router in prod as it uses client-side hash routing, which legacy browsers won't support and is bad for SEO (source: internet)
2. It looks bad having a hash in the URL. (Maybe just a personal feeling)

If you need any further clarification, please feel free to ping me. I have given a clear explanation in my comment above, but otherwise as of now this pr, passes all checks!
and I really feel this would be a good addition to the portfolio (I personally already use it in deployment)

**Comment by ashutosh1919** [comment]
@vjspranav Thanks for PR.

@allcontributors please add @vjspranav for code

**Pull request merged**
